src/p_maputl.c

```diff
@@ -419,13 +419,123 @@ boolean P_BlockLinesIterator(int x, int y, boolean func(line_t*))
 //
 // killough 5/3/98: reformatted, cleaned up
 
+boolean blockmapfix;
+extern boolean PIT_RadiusAttack(mobj_t* thing);
+
 boolean P_BlockThingsIterator(int x, int y, boolean func(mobj_t*))
 {
   mobj_t *mobj;
-  if (!(x<0 || y<0 || x>=bmapwidth || y>=bmapheight))
-    for (mobj = blocklinks[y*bmapwidth+x]; mobj; mobj = mobj->bnext)
-      if (!func(mobj))
-        return false;
+
+  if (x < 0 || y < 0 || x >= bmapwidth || y >= bmapheight)
+    return true;
+
+  for (mobj = blocklinks[y*bmapwidth+x]; mobj; mobj = mobj->bnext)
+    if (!func(mobj))
+      return false;
+
+  // Blockmap bug fix by Terry Hearst
+  // https://github.com/fabiangreffrath/crispy-doom/pull/723
+  // Add other mobjs from surrounding blocks that overlap this one
+  if (CRITICAL(blockmapfix))
+  {
+    if (func == PIT_RadiusAttack) // Don't do for explosions
```

Comment:
Not sure about MBF code, but have a look: https://github.com/JNechaevsky/inter-doom/blob/master/src/doom/p_maputl.c#L416-L420. 

In vanilla code `PIT_ChangeSector` is also necessary, otherwise crusher damage will be multiplied by amount of block mobj is taken. Can be checked on Doom II MAP06, take a look amount of blockmap blocks Spider is taking: [screenshot](https://user-images.githubusercontent.com/21193394/185755624-4e28cf5a-6503-4046-b61d-26bc3947f56a.png)

Reply:
Tested via a test map that the crushers kill large monsters a lot faster with the current implementation here, but it's a pretty easy fix.

Once the other changes are made, `if (func == PIT_RadiusAttack || func == PIT_ChangeSector)`

Reply:
As I suspected, thanks for checking, @SirBofu! I have been playing with this fix almost a year and can confirm that no other checks are needed. But however, again, I'm not familiar with MBF and maybe there are some extra stuff (was it called code pointers?) which must be taken into account. A_Mushroom comes to mind.

Reply:
Thanks everyone! Fixed in https://github.com/fabiangreffrath/woof/pull/705/commits/1e5c62388d1e90161f42f07ab1878c41881b2080